gossip/hub/handlers.go

```diff
@@ -626,6 +630,17 @@ func chargeTo(ctx context.Context) *trillian.ChargeTo {
 	return &trillian.ChargeTo{User: []string{qUser}}
 }
 
+// appendCharge adds the specified user to the passed in ChargeTo and and
+// returns the result. If the passed-in ChargeTo is nil, then a new one is
+// created with the passed in user and returned.
+func appendCharge(a *trillian.ChargeTo, user string) *trillian.ChargeTo {
+	if a == nil {
+		a = &trillian.ChargeTo{}
+	}
+	a.User = append(a.User, user)
```

Comment:
This function could be simplified slightly by using the proto [`GetUser()`](https://github.com/google/trillian/blob/HEAD/trillian_log_api.pb.go#L79) accessor. 

```suggestion
	a.User = append(a.GetUser(), user)
```

Reply:
Nice, done.

Reply:
(The net result is less efficient but neater code, which is a worthwhile tradeoff here.)

Reply:
Oh, one aside: the resulting code is more brittle to any changes to `ChargeTo` -- if any fields are added to its definition, they won't get copied across in the `appendCharge()` operation.